lib/formats/odata.js

```diff
@@ -70,8 +69,15 @@ const nextUrlFor = (limit, offset, count, originalUrl) =>
     : urlWithQueryParams(originalUrl, { $skip: (offset + limit), $top: null }));
 
 // Given a querystring object, returns an object of relevant OData options. Right
-// now that is only { wkt: Bool, expand: String }
-const extractOptions = (query) => ({ wkt: isTrue(query.$wkt), expand: query.$expand });
+// now that is only { wkt: Bool, expand: String, metadata: Array }
+const extractOptions = (query) => ({
+  wkt: isTrue(query.$wkt),
+  expand: query.$expand,
+  metadata: query.$select !== '*' && query.$select?.split(',')
+    .map(p => p.trim())
+    .filter(p => p.startsWith('__'))
+    .reduce((map, field) => ({ ...map, [field]: true }), {})
```

Comment:
I think it'd be a little better to initialize this as `Object.create(null)` instead of `{}`, since the keys are user-provided. Someone might have the bright idea of specifying `$select=__proto__`, in which case reasoning about this starts getting confusing. 😆 Or I think a `Set` would work well here!